compiler/rustc_privacy/src/lib.rs

```diff
@@ -507,11 +507,7 @@ impl EmbargoVisitor<'tcx> {
         }
         match def_kind {
             // No type privacy, so can be directly marked as reachable.
-            DefKind::Const
-            | DefKind::Macro(_)
-            | DefKind::Static
-            | DefKind::TraitAlias
-            | DefKind::TyAlias => {
+            DefKind::Const | DefKind::Static | DefKind::TraitAlias | DefKind::TyAlias => {
```

Comment:
Any specific reason for this change?
The old logic seemed correct to me.

Reply:
Yeah, there was, although looking back this may not be the right fix. The problem is that otherwise you start getting "macro has missing stability attribute" errors for non-exported macros. It would be possible to work around that, but I thought it would require reconstructing macro expectedness in atom `compiler/rustc_passes/src/stability.rs`. A potential alternative would be to only skip marking the item reachable for nonexported macros (in other words, handle macro expectedness here). I'm not sure which one of these solutions is actually correct.